File: UINavigationExtensionDemo/Details/Controllers/Basic/ViewController08_CustomBackButton.m
//
//  ViewController08_CustomBackButton.m
//  UINavigationExtensionDemo
//
//  Created by lidan on 2020/10/26.
//

#import "ViewController08_CustomBackButton.h"
#import "UIColor+RandomColor.h"

@interface ViewController08_CustomBackButton ()

@end

@implementation ViewController08_CustomBackButton

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (UIView *)ue_backButtonCustomView {
    UIButton *backButton = [UIButton buttonWithType:UIButtonTypeCustom];
    [backButton setTitle:@"Back" forState:UIControlStateNormal];
    [backButton setImage:[UIImage imageNamed:@"NavigationBarBack"] forState:UIControlStateNormal];
    [backButton setTitleColor:UIColor.customDarkGrayColor forState:UIControlStateNormal];
    return backButton;
}

@end

File: Example/NXNavigationExtensionSwiftDemo/Basic/ViewController08_CustomBackButton.swift
//
//  ViewController08_CustomBackButton.swift
//  NXNavigationExtensionSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class ViewController08_CustomBackButton: BaseTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override var nx_backButtonCustomView: UIView? {
        let backButton = UIButton(type: .custom)
        backButton.setTitle("😋", for: .normal)
        backButton.setImage(UIImage(named: "NavigationBarBack")?.withRenderingMode(.alwaysTemplate), for: .normal)
        backButton.setTitleColor(.customDarkGray, for: .normal)
        return backButton
    }

}
